esp/scm/ws_config2.ecm

```diff
@@ -0,0 +1,368 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+ESPstruct statusMsgType
+{
+    string msgLevel;                // message level ("info", "warning", "error", "fatal")
+    string nodeId;                  // node identifier that generated this status message
+    string attribute("");           // name of node's attribute generating the message ("" for the node itself)
+    string msg;                     // the message
+    ESParray<string, parentId> parentIdList;
+};
+
+ESPstruct statusType
+{
+    bool error(false);    // true if a message exsits in status (below) with a message level of error or higher
+    ESParray<ESPstruct statusMsgType, statusMsg> status;
+};
+
+
+ESPresponse PassFailResponse
+{
+    bool error(false);
```

Comment:
I wrote the interface to return an error flag set to true if there is an error. If instead it should pass a success flag, then I will have to make a number of changes. If there is a standard for this, please let me know. If there is no compelling reason (the UI is already using these interfaces) then can we leave it this way?

Reply:
Thinking back, the more common approach is to populate an error message (if err msg empty, success implied). 
Also, is this actually reporting an error? Errors should be reported via inline exceptions. This might actually (as the structure name implies) be reported success/failure right?
I might be splitting hairs, but if we're going to get this right, it has to be done now, we'll never go back and correct this type of issue later...

Reply:
PassFail response types were removed with the work done in jira issue HPCC-19216. Instead, the service will throw exceptions. The new issue is close to being finished and a PR will be issued shortly after this PR is merged.

Reply:
@kenrowland HPCC-19216's title doesn't seem related to this change, please make sure these change do indeed make it into that PR.

Reply:
All changes from HPCC-19216 relevant to requested changes on this PR have been implemented on this branch.